src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequences.tsx

```diff
@@ -35,8 +41,22 @@ const BlastInputSequences = () => {
     appendEmptyInputBox,
     setUncommittedSequencePresence
   } = useBlastForm();
+
   const shouldAppendEmptyInput = useAppSelector(getEmptyInputVisibility);
 
+  const blastFormContext = useContext(BlastFormContext);
+
+  const previousSequences = usePrevious(sequences);
+
+  useEffect(() => {
+    if (previousSequences?.length !== sequences.length) {
+      const sequenceFlags = sequences.map((sequence) => {
+        return checkSequenceValidity(sequence.value, sequenceType);
+      });
+      blastFormContext?.setSequenceValidityFlags(sequenceFlags);
+    }
+  }, [sequences]);
```

Comment:
I think, in this case, we can just rely on the array of `[sequences]` to indicate the change when the useEffect should run. There shouldn't be any need for the additional `if (previousSequences?.length !== sequences.length)` check that we are seeing on line 52.